### ISSUE TITLE
Contrib group membership request

### ISSUE THREAD

**Issue by goldmantx** [backdrop-ops/contrib#143]
I would like to be considered for contrib membership. I am more of a site builder but do tinker with the modules and have been Technical Reviewing some backdrop books for publication and want to get more involved in the community.

**Comment by klonos** [comment]
Hello there @goldmantx and welcome to Backdrop contrib. We generally do not give contrib membership to people not coding projects, but I am a live example of an exception to this rule.

At some point we said that we'd need to address this issue of people that need to contribute but not necessarily in code. I don't think we have actually gotten into it though.

I guess that what in general applied for my case would apply for your too, but I think we should wait on a decision and perhaps an official non-code contrib membership agreement drafted by either @quicksketch or @jenlampton. I've opened an issue for that.

**Comment by goldmantx** [comment]
Hi @klonos  - I guess I was thinking more along the lines of contributing in different ways like documentation and testing and I have to admit, didn't fully understand, but should have thought more about what I was asking. In another open source community I was involved in, we had to sign agreements to even be able to work on documentation. Sorry for any confusion.

**Comment by klonos** [comment]
Ah, ok then. If you need to work on documentation, then I guess first step is to register with  backdropcms.org account: https://backdropcms.org/user/register

If you do not have permission to create documentation there:

![backdropcms org-admin_bar-add_content](https://cloud.githubusercontent.com/assets/2423362/13577984/81985544-e4e9-11e5-9d39-0add2706747d.png)

...then I would file a separate issue over at https://github.com/backdrop-ops/backdropcms.org/issues

Don't mind if I close this one here?

**Comment by docwilmot** [comment]
I think @goldmantx may be talking about helping with testing modules and documenting code, rather than writing user docs on B.org. @goldmantx if you wanted to help with testing modules and documenting code, you'd need to push PRs to the contrib or Backdrop core repo but I dont know if you'd need to be a contrib author to do that. (Can you confirm @klonos?).

**Comment by goldmantx** [comment]
@docwilmot is correct, I am interested in testing modules, so unclear as to the process. Thanks for the clarification.

**Comment by klonos** [comment]
OK then, re-opening and waiting for either #145 or intervention from @quicksketch or @jenlampton because I feel that this is out of my jurisdiction. Besides, if such an official agreement is put together, I need to "sign" it because I am an exception right now AFAIK.

@goldmantx I hope you are ok with this and the whole wait does not put you off from contributing. It's just that I don't want to go over other people with more authority here.

**Comment by goldmantx** [comment]
@klonos not a problem at all. I am still on board and willing to help. I will wait.

**Comment by klonos** [comment]
Thanx. I trust it shouldn't be long :smile:

**Comment by quicksketch** [comment]
We've updated the project requirements in https://github.com/backdrop-ops/contrib/issues/145.

So now @goldmantx, we can get you into the contrib group. If you have further suggestions on improving the wording of this agreement, let's open another issue to refine it. To officially meet requirements, I'm nominating you to join the contrib group, as you have already joined the team meetings and participated in issues. :smile:

All we need is for you to agree to the following (even though only the last part really applies):

---

Backdrop Contributed Project Agreement
--------------------------------------

By joining the [Backdrop Contributed Project Group](https://github.com/backdrop-contrib) (Backdrop Contrib, for short) you
agree to the following:

1. You will not push changes to a repository for which you are not a current
   maintainer (even though joining the Backdrop Contrib group will grant you
   access to push to any project within).

1. You must agree to license your code contributions as GPLv2 or later.

1. Any project you create or maintain must include a copy of the [GPL v2 `LICENSE.txt` file](https://github.com/backdrop-ops/contrib/blob/master/examples/LICENSE.txt)
   in the root of your repository. The GPLv2 license applies to all code that
   directly interacts with parts of Backdrop licensed as GPLv2 or later.  See
   for the [Backdrop License FAQ](https://backdropcms.org/license) for a more
   detailed explanation.

1. You must confirm that you have the right to distribute any additional code,
   libraries, images, fonts or other assets written or created by any third
   party with code licensed as GPLv2 or later.

1. Any project you create or maintain must include a `README.md` file containing
  at the least the following:
  1. A description of the project
  1. Basic documentation
  1. License information for the project (GPL v2)
  1. License information for any additional assets (SIL OFL fonts, CC-SA images,
     etc)
  1. A list of the current maintainers for the Backdrop project
  1. Credits acknowledging past maintainers for the Backdrop or Drupal projects

  You may use this [example README.md](https://github.com/backdrop-ops/contrib/blob/master/examples/README.md) to get started.

1. You will work with the Backdrop Security Team to address any vulnerabilities
   in any project you create or maintain, if necessary.

1. Any project you create or maintain will have the GitHub issue tracker enabled
   for official communication.

1. If any of the above requirements are not met, your access to the Backdrop
   Contrib group -- including all projects and even those that you may have
   originally authored -- may be revoked.

1. If your project becomes abandoned and you do not respond to requests in the
   issue tracker within 2 weeks, your project may be modified by a Backdrop
   Contrib administrator without your explicit consent.

Additional notes for non-coders
-------------------------------

1. If an applicant is not (yet) a coder but would still like access to the
   contrib group (for example, for updating documentation or managing issue
   queues) the applicant must get at least one recommendation from another
   person who already has commit access to the contrib group.

1. Any non-coder applicant still agrees to the same contributed project author
   agreement as coders (above).

1. Any non-coder applicant needs to additionally promise that should they ever
   wish to author new code in the Backdrop contrib group, they will also undergo
   the same code spot-check process a new contributor needs to undergo at the
   time they add the code.

**Comment by goldmantx** [comment]
I agree, I agree!! :-) Thanks for getting that together @quicksketch and @jenlampton - I am excited to contribute and help out where I can.

**Comment by klonos** [comment]
FTR I also agree with all the new agreement points end especially the non-coder section.

**Comment by quicksketch** [comment]
Yay, great to have you helping @goldmantx! The invite is in the mail. You should now at least be able to categorize issues through labels on all contrib projects and you'll be ready to collaborate on any contrib modules that could use an extra hand. :smile:
